janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -498,58 +509,146 @@ private static Query numericQuery(String key, Cmp relation, Number value) {
         }
     }
 
+    // adapted from SolrIndex
+    private List<String> customTokenize(Analyzer analyzer, String fieldName, String value) {
+        final List<String> terms = new ArrayList<>();
+        try (CachingTokenFilter stream = new CachingTokenFilter(analyzer.tokenStream(fieldName, value))){
+            final TermToBytesRefAttribute termAtt = stream.getAttribute(TermToBytesRefAttribute.class);
+            stream.reset();
+            while (stream.incrementToken()) {
+                terms.add(termAtt.getBytesRef().utf8ToString());
+            }
+            return terms;
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+    }
+
+    private void tokenize(SearchParams params, KeyInformation.StoreRetriever information, String value, String key, JanusGraphPredicate janusgraphPredicate) {
+        final Parameter[] parameters = information.get(key).getParameters();
+        final Mapping mapping = ParameterType.MAPPING.findParameter(parameters, null);
+        final String analyzerString = mapping == null ? null : LuceneCustomAnalyzer.analyzerNameFor(parameters, mapping, null, null);
```

Comment:
Change to `final Analyzer analyzer` and `LuceneCustomAnalyzer.analyzerFor` (or whatever you rename it to)